hc-dna/zomes/social_context/src/methods.rs

```diff
@@ -5,14 +5,15 @@ use hdk::prelude::*;
 use crate::utils::{generate_link_path_permutations, get_link_permutation_by, LinkPermutation, get_wildcard};
 use crate::errors::{SocialContextError, SocialContextResult};
 use crate::{
-    GetLinks, LinkExpression, SocialContextDNA, UpdateLink, ACTIVE_AGENT_DURATION,
-    ENABLE_SIGNALS, ENABLE_TIME_INDEX, INDEX_STRAT, IndexStrategy, AgentReference
+    GetLinks, LinkExpression, SocialContextDNA, ACTIVE_AGENT_DURATION,
+    ENABLE_SIGNALS, ENABLE_TIME_INDEX, IndexStrategy, AgentReference,
+    AddLinkInput, UpdateLinkInput,
 };
 
 impl SocialContextDNA {
-    pub fn add_link(link: LinkExpression) -> SocialContextResult<()> {
+    pub fn add_link(input: AddLinkInput) -> SocialContextResult<()> {
```

Comment:
This break the api in ad4m, is this solution in the wrong direction? @jdeepee

Reply:
Nope this is perfect, since the ad4m language wraps the zome we dont have to worry about the DNA having the same interface as the LinkLanguage.

Just make sure that the when the ad4m language calls addLink we add the indexStrategy tag to the `input` for the addLink call. For now just have indexStrategy set to `FullWithWildCard` for all calls.